### ISSUE TITLE
Add method to save files directly from a filestream

### ISSUE THREAD

**Issue by josafatburmeister** [tweedegolf/storage-abstraction#2]
### Problem to solve

I would highly appreciate if the storage-abstraction interface would offer a method to save a [Readable Stream](https://nodejs.org/api/stream.html#stream_readable_streams) to a file. Right now, this is only possible by using a workaround like this: 

```javascript
const { Storage } = require("@tweedegolf/storage-abstraction")
const storageConfig = {..}

const storage = new Storage(storageConfig);
const readStream = createReadStreamSomeHow();

const chunks = [];
readStream.once('end', () => {
   const buffer = Buffer.concat(chunks);
   storage.addFileFromBuffer(buffer, 'targetPath');
});
readStream.on('data', (chunk) => {
   chunks.push(chunk);
});
```

Unfortunately this code is neither performant nor comprehensible.

### Proposed solution

I suggest to add method `addFileFromStream(fileStream: Readable,  targetPath: string): Promise<void>` to the storage-abstraction API. Using such a method could look like this:

```javascript
const { Storage } = require("@tweedegolf/storage-abstraction")
const storageConfig = {..}

const storage = new Storage(storageConfig);
const readStream = createReadStreamSomeHow();
await storage.addFileFromReadStream(readStream, 'targetPath'); 
```

**Comment by marlonbaeten** [comment]
@abudaan could you take a look at this?

**Comment by abudaan** [comment]
Implemented in 1.3.1
